tests/handlers/test_sliding_sync.py

```diff
@@ -3213,3 +3216,402 @@ def test_default_bump_event_types(self) -> None:
+            ),
+            state_deltas={},
+        )
+
+        # No changes
+        self.assertIsNone(to_persist)
+        self.assertEqual(added, StateFilter.none())
+
+    def test_simple_add_type(self) -> None:
+        """Test adding a type to the config"""
+        previous_required_state_map = {"type1": {"state_key"}}
+        request_required_state_map = {"type1": {"state_key"}, "type2": {"state_key"}}
+
+        to_persist, added = _required_state_changes(
+            user_id="@user:test",
+            previous_room_config=RoomSyncConfig(
+                timeline_limit=0,
+                required_state_map=previous_required_state_map,
+            ),
+            room_sync_config=RoomSyncConfig(
+                timeline_limit=0,
+                required_state_map=request_required_state_map,
+            ),
+            state_deltas={},
+        )
+
+        # We've added a type so we should persist the changed required state
+        # config.
+        assert to_persist is not None
+        self.assertDictEqual(to_persist, request_required_state_map)
+
+        # We should see the new type added
+        self.assertEqual(added, StateFilter.from_types([("type2", "state_key")]))
+
+    def test_simple_add_state_key(self) -> None:
+        """Test adding a state key to the config"""
+        previous_required_state_map = {"type": {"state_key1"}}
+        request_required_state_map = {"type": {"state_key1", "state_key2"}}
+
+        to_persist, added = _required_state_changes(
+            user_id="@user:test",
+            previous_room_config=RoomSyncConfig(
+                timeline_limit=0,
+                required_state_map=previous_required_state_map,
+            ),
+            room_sync_config=RoomSyncConfig(
+                timeline_limit=0,
+                required_state_map=request_required_state_map,
+            ),
+            state_deltas={},
+        )
+
+        # We've added a type so we should persist the changed required state
+        # config.
+        assert to_persist is not None
+        self.assertDictEqual(to_persist, request_required_state_map)
+        self.assertEqual(added, StateFilter.from_types([("type", "state_key2")]))
+
+    def test_simple_add_state_key_with_change(self) -> None:
+        """Test adding a state key to the config, and see some changed state"""
+        previous_required_state_map = {"type": {"state_key1"}}
+        request_required_state_map = {"type": {"state_key1", "state_key2"}}
+
+        to_persist, added = _required_state_changes(
+            user_id="@user:test",
+            previous_room_config=RoomSyncConfig(
+                timeline_limit=0,
+                required_state_map=previous_required_state_map,
+            ),
+            room_sync_config=RoomSyncConfig(
+                timeline_limit=0,
+                required_state_map=request_required_state_map,
+            ),
+            state_deltas={("type", "state_key"): "$event_id"},
```

Comment:
Should the `state_key` be matching anything in the previous/requested `required_state_map` to stress something specifically?

```suggestion
            state_deltas={("type", "state_key2"): "$event_id"},
```

If not, perhaps more clear that it isn't a typo but using:

```suggestion
            state_deltas={("type", "other_state_key"): "$event_id"},
```

Reply:
It doesn't matter for the `xxx_add` cases but I think this probably should be `state_key2` to match the state_key being added to the config.

This would matter in a `xxx_remove` test case

Reply:
Updated in https://github.com/element-hq/synapse/pull/17805